articles/040_stories.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: default
+title: Stories driving the development
+permalink: articles/stories.html
+abstract:
+  This article captures some stories which drive the direction of features in ROS.
+published: true
+author: '[Dirk Thomas](https://github.com/dirk-thomas)'
+categories: Overview
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+The article enumerates a few stories which sketch what will be possible with ROS in the future.
+The list is by no means exhaustive.
+
+## Push ROS into the hardware nodes
+
+A couple of design decisions in ROS 1 make it difficult to integrate hardware devices (e.g. a sensor or actuator) natively into the ROS graph.
+The master is a central entity in the ROS graph and needs to be started before any of the nodes.
+Also the communication between the nodes and the master is done using XML-RPC which poses a significant dependency when being implemented on small resource constraint systems / micro controllers.
```

Comment:
Might be worth working in the "why" there, which AFAIK is because XML is recursive (unbounded) and generally memory intensive.